app/src/main/java/space/taran/arknavigator/utils/DataStoreManager.kt

```diff
@@ -0,0 +1,36 @@
+package space.taran.arknavigator.utils
+
+import android.content.Context
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.first
+import javax.inject.Inject
+
+const val SHARED_PREFERENCES_KEY = "user_preferences"
+
+private val Context.preferencesDatastore by preferencesDataStore((SHARED_PREFERENCES_KEY))
```

Comment:
`SHARED_PREFERENCES_KEY` and `by preferencesDataStore((SHARED_PREFERENCES_KEY))` must be inside the `DataStoreManager` because `DateStore` will not be used without a `DataStoreManager`